src/openforms/locations/api/views.py

```diff
@@ -0,0 +1,62 @@
+from django.conf import settings
+from django.utils.module_loading import import_string
+from django.utils.translation import ugettext_lazy as _
+
+from drf_spectacular.types import OpenApiTypes
+from drf_spectacular.utils import OpenApiParameter, extend_schema
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from openforms.locations.api.serializers import (
+    GetStreetNameAndCityViewInputSerializer,
+    GetStreetNameAndCityViewResultSerializer,
+)
+
+
+class GetStreetNameAndCityView(APIView):
+    """
+    Get the street name and city when given a postcode and house number
+    """
+
+    authentication_classes = ()
```

Comment:
let's slap the permission class `AnySubmissionRequired` on this, otherwise anyone who finds this endpoint and has nothing to do with Open Forms can abuse this endpoint.